backend/fms_core/models/sample.py

```diff
@@ -279,23 +281,37 @@ def add_error(field: str, error: str):
             )
 
         if self.extracted_from:
-            if self.extracted_from.sample_kind.name in Sample.BIOSPECIMEN_TYPES_NA:
+            try:
+                ProcessSample = apps.get_model("fms_core", "ProcessSample")
+
+                sample_lineage = SampleLineage.objects.get(child_id=self.id)
+                protocol_name = ProcessSample.objects.get(id=sample_lineage.process_sample_id).protocol_name
+                if protocol_name == 'Extraction':
```

Comment:
this is some very strange way to implement this ... you have a property that should give you the extracted_from. You should implement the above in the property and remove it from here ... or maybe we want to remove the extracted_from property but going both way at once is not right.

Reply:
How would you access the protocol name from there? And you have to get the child_id, and not the parent, because the parent could result from multiple extractions/transfers/other processes.

Reply:
it should all be checked in the above: if self.extracted_from:
modify extracted_from to something like 
self.child_of.filter(parent_sample__child=self, parent_sample__process_sample__process__protocol__name__exact="Extraction").first()